src/pass/storage_rewrite.cc

```diff
@@ -569,12 +568,16 @@ class StoragePlanRewriter : public IRMutator {
                     make_const(sz.type(), alloc_type.lanes() - 1)) /
                   make_const(sz.type(), alloc_type.lanes());
             }
+            // transform to bits
+            auto sz_nbits = sz * (op->type.bits() * op->type.lanes());
```

Comment:
This part of logic was for dynamic memory allocation(allocation size unknown at compile time). I think we could try to keep the original restriction on only reuse same type for now. Since that covers most use-cases and avoids potential problem here. 

An improvement justified by usecase can be made later.

Reply:
Some comment on current code logic.  I just comment for your information:
 
The size calculation before this line makes the assumption of the element type matching. The combo_siza calculation which directly divide allocation bits will have rounding issues(need to round up).

We can safely ignore this(as per my last comment) by simply remove current change and add CHECK of the element type matching rule.

Reply:
add for this case below, 
```
       with ib.for_range(0, 10, name="j") as j:
 +            A = ib.allocate("float32", 200, name="A", scope="local.L0A")
 +            A[j] = 1.2
 +        with ib.for_range(0, 20, name="j") as j:
 +            B = ib.allocate("int16", 400, name="B", scope="local.L0A")
 +            B[j] = tvm.const(1, "int16")
```

Reply:
the case below is a statically know size(200, and 400) which won't go through this pass

Reply:
actually it is!

Reply:
OK then, please fix the logic according to my comments.

Reply:
specifically, the sz calculation still makes assumption of same type, the combo_size division is rounded down in current code, instead of rounded up

Reply:
ok